support-figma/extended-layout-plugin/src/shader.html

```diff
@@ -0,0 +1,226 @@
+        <button id="previewShaderButton" class="styled-button" style="width: 160px;"
+            onclick="runShader()">Preview</button>
+        <button id="insertImageButton" class="styled-button" style="width: 160px;" onclick="runShader()">Insert Running
+            Shader</button>
+    </div>
+
+    <div style="margin-top: 16px">
+        Shader examples are from:
+        <a target="_blank" href="https://shaders.skia.org/">https://shaders.skia.org/</a> and <a target="_blank"
+            href="https://shadertoy.com">https://shadertoy.com</a>
+    </div>
+
+    <div style="margin-top: 4px">
+        This tool is developed using Skia CanvasKit, please use
+        <a target="_blank" href="https://shaders.skia.org/">https://shaders.skia.org/</a> to debug.
+    </div>
+</div>
+<!-- END OF SHADER PLUGIN UI -->
+
+<script>
+    let canvas = document.getElementById("shaderPreview");
+    let shaderCodeInput = document.getElementById("shaderCodeInput");
+
+    let insertButton = document.getElementById("insertImageButton");
+    let currentShader = null;
+
+    let currentImageSnapshot = null;
+
+    function clearShaderCode() {
+        shaderCodeInput.value = "";
+        currentShader = null;
+    }
+
+    function loadShaderCode(shader) {
+        parent.postMessage({
+            pluginMessage: {
+                msg: 'loadShaderCode',
+                shader: shader,
+            }
+        }, '*');
+    }
+
+    async function runShader() {
+        CanvasKitInit({
+            locateFile: (file) =>
+                "https://unpkg.com/canvaskit-wasm@latest/bin/" + file,
+        }).then((CanvasKit) => {
+            // Revisied from skia playground https://shaders.skia.org/
+            const shaderWidth = 256;
+            const shaderHeight = 256;
+
+            let surface = CanvasKit.MakeCanvasSurface(canvas.id);
+            if (!surface) {
+                throw "Could not make surface";
+            }
+            const skcanvas = surface.getCanvas();
+            const paint = new CanvasKit.Paint();
+            const startTimeMs = Date.now();
+            let mouseClickX = 0;
+            let mouseClickY = 0;
+            let mouseDragX = 0;
+            let mouseDragY = 0;
+            let lastMousePressure = 0;
+
+            const prog = `
+// Inputs supplied by shaders.skia.org:
+uniform float3 iResolution;      // Viewport resolution (pixels)
+uniform float  iTime;            // Shader playback time (s)
+uniform float4 iMouse;           // Mouse drag pos=.xy Click pos=.zw (pixels)
+` + shaderCodeInput.value;
+
+            currentShader = prog;
+
+            const effect = CanvasKit.RuntimeEffect.Make(prog);
+
+            function drawFrame(canvas) {
+                canvas.clear(CanvasKit.BLACK);
+                const iTime = (Date.now() - startTimeMs) / 1000;
+                const uniforms = [
+                    shaderWidth,
+                    shaderHeight,
+                    1, // iResolution
```

Comment:
Can this be changed in the plugin? Would it be useful?

Reply:
iResolution has x,y,z. You are asking the z being 1? I tried to change it to something else, but it doesn't make any differences for those shader examples I put in this pr which only use the x,y coordinates.

I don't really know how z-depth works. The Skia CanvasKit may not even support z depth(this is the answer from [gemini](https://gemini.google.com/corp/share/9b38fa505ca2) if I am asking it the right question).

The shader width and height can change.